the-collab-lab/tcl-22-smart-shopping-list
# Fix first purchase estimate calculation

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



[Previous fix](https://github.com/the-collab-lab/tcl-22-smart-shopping-list/pull/44) is not working as expected because the sorting logic is based on previous purchase dates.

This PR adds a new field `dateAdded` to the database to store the date when the user added the item and set its first estimated purchase date. This date is then used to put the item into one of the list segments based on how many days are left until its first estimated purchase.

## Related Issue


Closes #43
(for good)

## Acceptance Criteria



* When a new item is added, it shows up in [Firebase](https://console.firebase.google.com/u/2/project/tcl-22-shopping-list/firestore/)
* The new item in Firebase has the field `purchaseEstimates` containing 1 element that is equal to the frequency set by the user in the Add Groceries form and the field `dateAdded` that contains the current date.
* The new item is put into the correct segment of the grocery list.

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓  | :bug: Bug fix              |

## Updates

### Before



https://www.loom.com/share/39708b3a417847b4aaff28961684ba02

### After



https://www.loom.com/share/e8751fb42b7c4b3a82e6ed7dc606f373

## Testing Steps / QA Criteria



To test the feature, use one of the following ways:

1. Follow the [preview link](https://tcl-22-shopping-list--pr58-dt-new-item-estimate-zbiqo5e2.web.app)
2. Launch the app locally from your terminal:
* Pull down this branch with `git pull origin dt-new-item-estimate-fix` and check that branch out with `git checkout origin dt-new-item-estimate-fix`
* Run `npm start`  to launch the app in your browser at [http://localhost:3000](http://localhost:3000)

When on application:

* In Add Items view, add a new item and notice the frequency you are setting it to.
* In [Firebase](https://console.firebase.google.com/u/2/project/tcl-22-shopping-list/firestore/data), your item's `purchaseEstimates` should have the value that you set and its `dateAdded` should be the current date.
* The item should appear in the correct segment of the grocery list based on the value you set.
